feat(debug): add debug logging in configure

beekhc opened the pull request:
## Which problem is this PR solving?

The SDK currently doesn't print debug information when configured.

## Short description of the changes

This adds debug output consistent with other platforms. It was copied almost verbatim from https://github.com/honeycombio/honeycomb-opentelemetry-swift/pull/19

## How to verify that this has the expected result

Run the smoke tests in Android Studio and verify the output in logcat.

beekhc committed:
add debug logging in configure

martin308 commented on core/src/main/java/io/honeycomb/opentelemetry/android/HoneycombDebug.kt:
```diff
@@ -0,0 +1,12 @@
+    if (options.debug) {
+        print("🐝 Honeycomb SDK Debug Mode Enabled 🐝")
+
+        print("API Key configured for traces: ${options.tracesApiKey}")
```
does this imply there are api keys for all the signals? If so should we also print them?

beekhc replied:
okay, I changed it to just print the whole options object, and verified in logical that it's reasonable.

I also changed it to `Log.d`, because it turns out `print` doesn't show up in logcat.

please take another look

beekhc committed:
feat: add BaggageSpanProcessor (#26)

This adds a BaggageSpanProcessor to the Android SDK, per Honeycomb
convention.

The smoke test was updated to add a baggage attribute to the custom
span.

MustafaHaddara committed:
feat: add render instrumentation (#20)

## Which problem is this PR solving?
Adds manual instrumentation for rendering times of Composables in
Jetpack Compose

## Short description of the changes
heavily inspired by
https://github.com/honeycombio/honeycomb-opentelemetry-swift/pull/20


Multiple re-renders do get nested under a single "Created" Span, but
otherwise the spans look identical to the iOS spans:


![image](https://github.com/user-attachments/assets/50460236-9649-4264-b457-b4be9d1eca62)

## How to verify that this has the expected result
- [x] traces appear in honeycomb with the correct values (see screenshot
above)
- [x] smoke tests

beekhc committed:
change to Log and print _all_ options

beekhc committed:
Merge branch 'main' into beeklimt.debug

beekhc commented:
Given that it's a small change, is consistent with Swift's new debug, and I'm applying your feedback, I'm gonna go ahead and submit this, so that we can clear it off the board today.
